### PR TITLE
[enzyme_v3.x.x]: add simulateError to enzyme_v3.x.x

### PR DESCRIPTION
**Pull request by HsuTing**
- Links to documentation: https://github.com/airbnb/enzyme/blob/944f9e098bb1978c750525416a18a58c80fbe126/docs/api/ReactWrapper/simulateError.md
- Link to GitHub or NPM: https://github.com/airbnb/enzyme
- Type of contribution: addition

### CODE REVIEW

File: definitions/npm/enzyme_v3.x.x/flow_v0.53.x-/enzyme_v3.x.x.js [hunk 1]
```diff
@@ -49,6 +49,7 @@ declare module "enzyme" {
     prop(key: string): any,
     key(): string,
     simulate(event: string, ...args: Array<any>): this,
+    simulateError(error: any): this,
```
Comment (pascalduez) [inline definitions/npm/enzyme_v3.x.x/flow_v0.53.x-/enzyme_v3.x.x.js]: Isn't it a bit sad to use `any` here? 
The method calls `componentDidCatch` under the hood.

Comment (HsuTing) [reply definitions/npm/enzyme_v3.x.x/flow_v0.53.x-/enzyme_v3.x.x.js]: In [docs](https://github.com/airbnb/enzyme/blob/944f9e098bb1978c750525416a18a58c80fbe126/docs/api/ReactWrapper/simulateError.md#arguments), the type of error is `any`.

I had traced the [code](
https://github.com/airbnb/enzyme/blob/02d63378a36a74c4e76f74a40404b218b9905e66/packages/enzyme-adapter-utils/src/Utils.js#L281), it call `componentDidCatch` directly. In `flow`, the type of error is `Error` in [componentDidCatch](https://github.com/facebook/flow/blob/master/lib/react.js#L77-L82).

Maybe `Error` is more better?

Comment (pascalduez) [reply definitions/npm/enzyme_v3.x.x/flow_v0.53.x-/enzyme_v3.x.x.js]: I would say so yes.

Comment (HsuTing) [reply definitions/npm/enzyme_v3.x.x/flow_v0.53.x-/enzyme_v3.x.x.js]: OK

**Pull request merged**